docs/howto/object-storage/s3/object-lock.md

```diff
@@ -0,0 +1,271 @@
+# Object lock
+
+Object lock is a feature in S3 that enables you to lock your objects
+to prevent them from being deleted or overwritten. This feature provides
+a storage solution based on the **Write Once, Read Many (WORM)** model,
+which can be helpful when you want to maintain data integrity,
+comply with regulations, or retain data for a specific period.
+
+> Are you looking to set the expiration/retention time for your
+> objects? Then you should check our [Object expiry](expiry.md) guide.
+
+## Object lock modes
+
+Object lock offers two methods for managing object retention:
+**Retention periods** and **Legal hold**.
+
+### Retention periods
+
+Retention periods specify the length of time that an object
+remains locked and protected from being overwritten or deleted.
+When you set a retention period, the object is safeguarded for
+the specified time period. You can set the retention period in
+either days or years, with a **minimum of one day** and
+no maximum limit.
+
+In addition to retention periods, you can choose the retention mode
+that applies to your objects (either Governance or Compliance). In
+{{brand}}, the only supported retention mode is **Compliance**.
+
+Compliance mode is recommended when storing compliant data,
+as it prevents objects from being overwritten or deleted
+by any user. If you configure an object with this mode, you
+cannot shorten or change its retention period, ensuring that
+the data remains **secure** and **compliant** with regulatory
+requirements.
+
+### Legal hold
+
+Legal hold is a feature designed for situations when you are uncertain
+about the length of time you need to retain an object. It can be
+enabled/disabled for any object in a locked bucket, regardless of the
+lock configuration, object retention, or age.
+
+This feature provides the same level of protection as a retention period
+but with no fixed expiration date. Instead, a legal hold will remain in
+effect until you remove it explicitly.
+
+## Prerequisites
+
+In order to manage object lock, you must install `aws` or `mc`, and [configure it with working credentials](credentials.md).
+
+You cannot (reliably) use `s3cmd` to manage this functionality.
+
+## Enabling object lock
+
+To use the object lock feature, you must first create a new bucket.
+
+> Existing buckets cannot have object lock enabled.
+> If you require assistance to enable object lock on an existing bucket,
+> please contact our support team.
```

Comment:
```suggestion
> please get in touch with our support team.
```

Reply:
Actually, on that note. It just now occurs to me that that paragraph is kind of a contradiction, because if "existing buckets *cannot* have object lock enabled" (emphasis mine), then it doesn't make sense to contact support either.

So we either need to rephrase this to reflect that *customers* cannot enable object lock on an existing bucket (but we can), or else drop the support reference altogether.

@revij, can you clarify please?

Reply:
The idea here is that if a customer has a large bucket and wants to have object locking enabled, we can create that bucket ourselves and copy over the data, otherwise the customer will have to pay 2x the ingress fee to create a new bucket and upload everything again.

Reply:
No worries, fixed in [97463f3](https://github.com/citynetwork/docs/pull/135/commits/97463f3880f90bf0587ad81e2e0ae94cd5517c67) :)